### ISSUE TITLE
[BUG/ISSUE] ND51 output in nested simulation.

### ISSUE THREAD

**Issue by Rainbow1994** [geoschem/geos-chem#206]
# Report a GEOS-Chem bug or technical issue

## Describe the bug
I am using v12.5.0 to run a nested Asia simulation. when I turned on ND51 output menu in input.geos, and modified parameters according to  http://wiki.seas.harvard.edu/geos-chem/index.php/The_input.geos_file#ND51_and_ND51b_diagnostics, an error occurred while running.  I wonder if I typed wrong index of Asia, and the IMIN, IMAX ... of region are indics of the extent of sub region such as Asia, or indics of the global grid. 

## Error messages

![Picture1](https://user-images.githubusercontent.com/29177864/73186019-001a5380-4117-11ea-9272-763d8004c649.png)

## Input and log files to attach
For more info, see: http://wiki.geos-chem.org/Submitting_GEOS-Chem_support_requests
 - The input.geos file

[input.geos.txt](https://github.com/geoschem/geos-chem/files/4117717/input.geos.txt)

**Comment by yantosca** [comment]
Thanks for writing.  One thing you can do is to check the values of State_Grid%NX and State_Grid%NY against the ND51 menu settings.

```
%%% ND51 MENU %%%       :
Turn on ND51 diagnostic : T
LT avg timeseries file  : /scratch/local/shzhu/NOX_GC/TEST/asian_ensemble_run/Diagn_files/ND51/ts_satellite.ST000.EN0001-EN0008.YYYYMMDD.bpch
Output as HDF5?         : F
Tracers to include      :   1  2  3  4  5  6  7  8  9  10  11  12  13  14  15  
GMT Hour for disk write :   0
Avg Period [LT hours]   :  10  12
IMIN, IMAX of region    :   1 145
JMIN, JMAX of region    :   1 133
LMIN, LMAX of region    :   1 47
```

Also, be sure that you are using the correct 12.5.0 run directory.  If you are using a run directory from an older version of the unit tester, there might be a mismatch in the line count between what is expected to be read in and what is actually read in.  You can also take a look in the READ_ND51_MENU routine within GeosCore/input_mod.F to see if each READ_ONE_LINE call corresponds to each line of the input.geos file.  If there is a mismatch, you should find it easily.

**Comment by Rainbow1994** [comment]
Thanks for your quick reply. I checked input_mod.F right now, source code and input.geos match to each other. When I checked State_Grid%NX and State_Grid%NY in diag51_mod.f such as pictures below, it turned out I0 = State_Grid%XMinOffset value is wrong, I wanna to know how to the define the value in config files.
![image](https://user-images.githubusercontent.com/29177864/73194875-bf293b80-4124-11ea-92bf-620671776e19.png)
![image](https://user-images.githubusercontent.com/29177864/73194560-475b1100-4124-11ea-93ff-e12f19813944.png)

**Comment by yantosca** [comment]
Is your region trying to wrap around the date line?  I wonder if that is the issue.

**Comment by Rainbow1994** [comment]
I checked the source code, maybe there are some errors in diag51_mod.F. Variable State_Grid%XMinOffset is the boundary gird index of area in default global grid with the same resolution. 
![image](https://user-images.githubusercontent.com/29177864/73256385-b5054c80-41b9-11ea-8c50-c4414bb86c84.png)
However, in diag51_mod to check ND51_IMIN, it need to achieve buffer zone shown in input.geos.
![image](https://user-images.githubusercontent.com/29177864/73256078-2ee90600-41b9-11ea-8ecb-af054dc35392.png)
they are read in input_mod.f as State_Grid%NorthBuffer.
![image](https://user-images.githubusercontent.com/29177864/73256148-5344e280-41b9-11ea-9535-b6458bcb2766.png)
I need your further verification!  Many thanks in advance~

**Comment by yantosca** [comment]
Can you add the log file with the output of GRID MENU (and not just the input.geos).  That might be instructive.

Also, the buffer settings are only for transport (TPCORE).  But you should probably not rely on any data within the transport window (i.e. the real window will be 3 boxes in from each edge).

**Comment by Rainbow1994** [comment]
Here is the output.log file !
[output.log](https://github.com/geoschem/geos-chem/files/4124192/output.log)

**Comment by Rainbow1994** [comment]
Are there some methods to fix the problem?

**Comment by yantosca** [comment]
I finally discovered what was going on.   The recent updates to FlexGrid in GEOS-Chem 12.4.0 and 12.5.0 changed how I0 and J0 were defined.  Therefore, we no longer need to use them in the error traps.

In other words, we changed:

```F90
      IF ( Input_Opt%ND51_IMIN+I0 < 1       .or.
     &     Input_Opt%ND51_IMIN+I0 > State_Grid%NX ) THEN
... etc...
```
to
```F90
      IF ( Input_Opt%ND51_IMIN < 1       .or.
     &     Input_Opt%ND51_IMIN > State_Grid%NX ) THEN
...etc...
```

This is implemented in https://github.com/geoschem/geos-chem/commit/24e6f723cb4cc408e74fab9cd3510aeda4dc4fa1, which will be added to GEOS-Chem 12.7.1,
